gravitee-apim-gateway/gravitee-apim-gateway-http/src/main/java/io/gravitee/gateway/jupiter/http/vertx/VertxHttpServerResponse.java

```diff
@@ -190,10 +190,10 @@ public Completable end() {
         );
     }
 
-    private void cacheChunks(Flowable<Buffer> chunks) {
+    private void cacheChunks(Flowable<Buffer> chunks, boolean force) {
         this.chunks = chunks;
 
-        if (cached.compareAndSet(false, true)) {
+        if (force || cached.compareAndSet(false, true)) {
```

Comment:
We have totaly remove this atomicboolean last week or two. Do we still need it ?

Reply:
AFAIK it has not been removed yet (maybe on your branch). I've put it to make sure it remains optimal. I suggest to leave it like this until it is removed when the development you mentioned will be merge.